tests/driver.sh

```diff
@@ -6,58 +6,47 @@ readonly SHECC="$PWD/out/shecc"
 
 function try() {
     local expected="$1"
-    local input="$2"
+    local expected_output="$2"
+    local input="$3"
 
     local tmp_in="$(mktemp --suffix .c)"
     local tmp_exe="$(mktemp)"
     echo "$input" > "$tmp_in"
     "$SHECC" -o "$tmp_exe" "$tmp_in"
     chmod +x $tmp_exe
-    $TARGET_EXEC "$tmp_exe"
+
+    local output=''
+    output=$($TARGET_EXEC "$tmp_exe")
     local actual="$?"
 
-    if [ "$actual" = "$expected" ]; then
-        echo "$input => $actual"
-    else
+    if [ "$actual" != "$expected" ]; then
         echo "$input => $expected expected, but got $actual"
         echo "input: $tmp_in"
         echo "executable: $tmp_exe"
         exit 1
+    elif [ "$output" != "$expected_output" ]; then
+        echo "$input => $expected_output expected, but got $output"
+        echo "input: $tmp_in"
+        echo "executable: $tmp_exe"
+        exit 2
+    else
+        echo "$input"
+        echo "exit code => $actual"
+        echo "output => $output"
     fi
 }
 
 function try_() {
-    local expected="$1"
-    local input="$(cat)"
-    try "$expected" "$input"
-}
-
-function try_output() {
     local expected="$1"
     local expected_output="$2"
     local input="$(cat)"
-    
-    local tmp_in="$(mktemp --suffix .c)"
-    local tmp_exe="$(mktemp)"
-    echo "$input" > "$tmp_in"
-    "$SHECC" -o "$tmp_exe" "$tmp_in"
-    chmod +x $tmp_exe
-    output=$($TARGET_EXEC "$tmp_exe")
-    
-    if [ "$output" = "$expected_output" ]; then
-        echo "$input => $output"
-    else
-        echo "$input => $expected_output expected, but got $output"
-        echo "input: $tmp_in"
-        echo "executable: $tmp_exe"
-        exit 1
-    fi
+    try "$expected" "$expected_output" "$input"
 }
 
 function items() {
     local expected="$1"
     local input="$2"
-    try "$expected" "int main(int argc, int argv) { $input }"
+    try "$expected" "" "int main(int argc, int argv) { $input }"
```

Comment:
It is not elegant. Instead, we can obtain the number of arguments passed to a shell function.
See https://linuxhint.com/check-the-number-of-arguments-in-bash/